Source/santametricservice/SNTMetricServiceTest.m

```diff
@@ -112,4 +125,51 @@ - (void)testWritingRawJSONFile {
 
   XCTAssertEqualObjects(validMetricsDict, parsedJSONData, @"invalid JSON created");
 }
+
+- (void)testWritingJSON {
+  NSURL *url = [NSURL URLWithString:@"http://localhost:9444"];
+  OCMStub([self.mockConfigurator exportMetrics]).andReturn(YES);
+  OCMStub([self.mockConfigurator metricFormat]).andReturn(SNTMetricFormatTypeRawJSON);
+  OCMStub([self.mockConfigurator metricURL]).andReturn(url);
+
+  self.mockSession = OCMClassMock([NSURLSession class]);
+  self.mockSessionDataTask = OCMClassMock([NSURLSessionDataTask class]);
+  self.mockMOLAuthenticatingURLSession = OCMClassMock([MOLAuthenticatingURLSession class]);
+
+  OCMStub([self.mockMOLAuthenticatingURLSession alloc])
```

Comment:
We need to stub this out or stub out NSURLSession's `sessionWithConfig:delegate:delegateQueue`. Since that's what `MOLAuthenticatingURLSession` is using for session under the hood. 

I went with this to follow the rule of least surprise as it matches the code being tested.